seclea_ai/svc/api/collection/dataset.py

```diff
@@ -26,23 +26,23 @@ def post_dataset(
     @param organization_pk:
     @param name:
     @param metadata:
-    @param dataset_hash:
+    @param dataset_pk:
     @param parent_dataset_hash:
     @param delete: delete dataset file
     @return:
     """
-    dataset_queryparams = {
-        "project": project_pk,
-        "organization": organization_pk,
-        "name": name,
-        "metadata": json.dumps(metadata),
-        "hash": str(dataset_hash),
-        "parent": parent_dataset_hash,
+    dataset_queryparams = {"project": project_pk, "organization": organization_pk}
+    dataset_obj = {
+        "project": (None, project_pk),
+        "name": (None, name),
+        "metadata": (None, json.dumps({"Test": 1})),
+        "hash": (None, str(dataset_pk)),
+        "parent": (None, parent_dataset_hash),
+        "dataset": ("dataset_name", open(dataset_file_path, "rb")),
```

Comment:
Better to wrap this all in a context manager that opens the file. That way it will be closed when finished with.